webull.py

```diff
@@ -179,6 +179,122 @@ def get_trade_token(self, password='') :
         else :
             return False
 
+    def alerts_list(self):
+        '''
+        Get alerts
+        '''
+        headers = self.build_req_headers()
+        url = 'https://userapi.webullbroker.com/api/user/warning/v2/query/tickers'
```

Comment:
Not asking you to do it, but wanted your opinion on defining the different groups of api base urls and using them instead of repeating so much throughout this file...
so instead of what you have here, we could do...
in urls.py:
base_user_url = 'https://userapi.webull.com/api'
def alerts_list():
    return f'{base_user_url}/user/warning/v2/query/tickers'
...etc.
and in webull.py:
import urls
requests.get(urls.alerts_list(), headers=headers)

Or just define them (there will be 7 unique ones after adding paper account functions) globally in webull.py and replace throughout it without adding a new file.

Reply:
How about keeping the endpoints in a dict of hosts and routes?

endpoints = {'user_host': 'https://userapi.webull.com',
    'user_routes': {
        'user': '/api/user',
        'tickers': '/api/user/warning/v2/query/tickers'
    },
    'quotegw_host': 'https://quotes-gw.webullbroker.com',
    'quotegw_routes: {
        'list': '/api/quote/option/query/list'
    },
    'trade_host': ''https://tradeapi.webulltrade.com',
    'trade_routes': {
        'place_order': '/api/trade/v2/option/placeOrder/{self.account_id}',
        'replace_order': '/api/trade/v2/option/replaceOrder/{self.account_id}',
    },
}

Reply:
so it'd be something like
requests.get(f'{endpoints['user_host']}{endpoints['user_routes']['user']}', headers=headers)

I'm not sure what's best. To me that is not pretty or convenient to read. If there is any official APIs we could mock (alpaca or ally, for example), I'd be all for that. They seem to typically only have 1 base request endpoint for the public though. :( Do you have any examples? I was just going off of what I've seen in other algo projects.